src/FsToolkit.ErrorHandling/Seq.fs

```diff
@@ -1,19 +1,37 @@
-namespace FsToolkit.ErrorHandling
-
 [<RequireQualifiedAccess>]
-module Seq =
-
-    let sequenceResultM (xs: seq<Result<'t, 'e>>) : Result<'t seq, 'e> =
-        let rec loop xs ts =
-            match Seq.tryHead xs with
-            | Some x ->
-                x
-                |> Result.bind (fun t -> loop (Seq.tail xs) (t :: ts))
-            | None ->
-                Ok(
-                    List.rev ts
-                    |> List.toSeq
-                )
-
-        // Seq.cache prevents double evaluation in Seq.tail
-        loop (Seq.cache xs) []
+module FsToolkit.ErrorHandling.Seq
+
+let sequenceResultM (xs: seq<Result<'t, 'e>>) : Result<'t[], 'e> =
```

Comment:
note we return an array for consistency with sequenceResultA
if people want arrays or allocations, List.sequenceResultM covers that

Reply:
I'd prefer not to make breaking changes here unless you want to retarget for [v5](https://github.com/demystifyfp/FsToolkit.ErrorHandling/pull/249).

Reply:
Yeah either way I'm fine, since it's not time sensitive, lets retarget for v5.

Reply:
rebased on v5